ops/framework.py

```diff
@@ -1010,6 +1045,70 @@ def __get__(self, parent, parent_type=None):
                 self.__class__.__name__, parent_type.__name__))
 
 
+class ConfigHistoryObserver(Object):
+    """A class used to store config history for current application."""
+    _stored = StoredState()
+
+    def __init__(self, parent):
+        super().__init__(parent, None)
+        self._keys = list(self.model.config.keys())
+        self._depth = -1
+        self._stored.set_default(history={key: [self.model.config[key]] for key in self.keys})
+
+    @property
+    def history(self) -> 'StoredDict':
+        """Config history."""
+        return self._stored.history
+
+    @property
+    def keys(self) -> typing.List[str]:
+        """List of tacked keys."""
+        return self._keys
+
+    @keys.setter
+    def keys(self, keys: typing.List[str]) -> None:
+        """Set list if tacked keys."""
+        _keys = []
+        for key in keys:
+            if key not in self.model.config.keys():
+                raise ValueError("Key {} was not found in application config.".format(key))
+
+            _keys.append(key)
+
+        self._keys = keys
+
+    def set_depth(self, depth: int) -> None:
+        """Set depth of tacked history."""
+        if not isinstance(depth, int) and (depth == -1 or 1 <= depth):
+            raise ValueError("Depth must be integer from range [1, inf). [inf <==> -1]")
+
+        self._depth = depth
+
+    def _get_last_value(self, key: str) -> typing.Any:
+        """Get last value for certain key."""
+        key_history = self._stored.history.get(key, [])
+        if not isinstance(key_history, (list, StoredList)) or len(key_history) == 0:
+            raise RuntimeError("Could not get last value for %s", key)
```

Comment:
KeyError and/or IndexError might be a better fit?

Reply:
Unless the user changes `_stored.history` manually (maybe we could prevent this?), it should never happen that the value is not a list/StoredList or that it is an empty list. That is why I chose `RuntimeError`, but I have no issue to change it to `KeyError`/`IndexError`.